SwiftyStoreKit/PaymentQueueController.swift

```diff
@@ -91,6 +94,26 @@ class PaymentQueueController: NSObject, SKPaymentTransactionObserver {
         paymentQueue.remove(self)
     }
 
+	var unhandledTransactions: [Purchase] {
+		guard let completeTransactions = completeTransactionsController.completeTransactions else {
+			print("SwiftyStoreKit.completeTransactions() should be called once when the app launches. Returning no results")
+			return []
+		}
+		let purchases = paymentQueue.transactions.flatMap { transaction -> Purchase? in
+			guard transaction.transactionState != .purchasing else {
+				return nil
+			}
+			return Purchase(
+				productId: transaction.payment.productIdentifier,
+				quantity: transaction.payment.quantity,
+				transaction: transaction,
+				originalTransaction: transaction.original,
+				needsFinishTransaction: !completeTransactions.atomically
```

Comment:
That's  very good point.

Maybe I should explain a bit about our use case. Our app [Carousell](https://appsto.re/sg/ttXSG.i) is a C2C platform that let user sell their items. One user can have as many items as they want, each item belongs to a category. The In App Purchase that we're selling is Consumables which let user Bump their item to top of the category. Each category has different pricing to bump. That essentially means we have one-to-many mapping `iTunesConnect Price ID` -> `Our internal ProductID`. 

The way we handle unfinished transactions (due to previous failed request to fulfil) is that we present an UI to user to let them initiate a call to our backend to fulfil it. Then we remove it from the queue after receiving a success response from our backend. Due to the one-to-many mapping, user can initiate this fulfilment on any product that has the same pricing as the unfinished transaction.

To achieve this, we need a `SKPaymentTransaction` object to call `SwiftyStoreKit.finishTransaction()` on, when the fulfilment is successful

Do you have any suggestion? Thanks !